storage/remote/queue_manager_test.go

```diff
@@ -356,22 +356,27 @@ func TestReshardRaceWithStop(t *testing.T) {
 
 	cfg := config.DefaultQueueConfig
 	mcfg := config.DefaultMetadataConfig
+	exited := false
```

Comment:
Using a bool here will cause a race (looks like it is properly failing in the tests as well). You could use a channel instead or obtain `h.Lock()` before modifying exit.

Reply:
thanks for pointing out, could you review the latest change? @csmarchbanks